docs/api.yaml

```diff
@@ -0,0 +1,641 @@
+        500:
+          $ref: "#/components/responses/500"
+
+  /group:
+    patch:
+      tags:
+        - "Student"
+      summary: "Attribute a group to a student by updating his group"
+      description: ""
+      operationId: "attributeGroup"
+      requestBody:
+        description: "The student and the group to attribute him"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Student'
+      responses:
+        200:
+          description: "The updated Student"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Student'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /courses/student:
+    post:
+      tags:
+        - "Course"
+      summary: "Enroll a student to a course"
+      description: "Create a new enrolled course for a student"
+      operationId: "enrollCourse"
+      requestBody:
+        description: "The student and the course to attribute him"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/EnrolledCourse'
+      responses:
+        200:
+          description: "The created EnrolledCourse"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/EnrolledCourse'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /courses/teacher:
+    post:
+      tags:
+        - "Course"
+      summary: "Attribute a course to a teacher"
+      description: "Create a new relation between a teacher and a teaching unit"
+      operationId: "attributeTeacherCourse"
+      requestBody:
+        description: "The teacher and the teaching unit to attribute him"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TaughtCourse'
+      responses:
+        200:
+          description: "The created TaughtCourse"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TaughtCourse'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /activation:
```

Comment:
No. PATCH the Student resource, and the PATCH-ed attribute is status. That is: `PATCH /students/{id}` with payload `{'status' : 'ENABLED'}`

Reply:
The parameter id is not inside the path but the query.

Reply:
Thinking again about it, do not put the id neither in the path nor in the parameters. Doing so will prevent batch patching (ie, patching a whole list of students). You should put the id the in payload, attached to each student we want to patch.